src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,72 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static String RESULT = "result";
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final int OPERATIONS_INDEX = 0;
+    private static final int OPERATIONS_INDEX_VALUE = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String readData = readFromFile(fromFileName);
+        String resultReport = createReport(readData);
+        writeFile(toFileName, resultReport);
+    }
+
+    private String readFromFile(String fromFileName) {
+        File fromFile = new File(fromFileName);
+        StringBuilder stringBuilder;
+        int value;
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFile))) {
+            stringBuilder = new StringBuilder();
+            value = bufferedReader.read();
+            while (value != -1) {
+                stringBuilder.append((char) value);
+                value = bufferedReader.read();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file" + e);
+        }
+        return stringBuilder.toString();
+    }
+
+    private void writeFile(String toFileName, String resultReport) {
+        File toFile = new File(toFileName);
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFile))) {
+            bufferedWriter.write(resultReport);
+            bufferedWriter.flush();
```

Comment:
This is redundant, becouse anyway you call method close in the try-with-resources block
```suggestion
```